### PR TITLE
build: updated solid-social

### PR DESCRIPTION
**Pull request by high1**
- updated solid-social to 0.8.0, which had a large PR that updated it's build system to make it compatible with SSR by @modderme123 (you :rock: )
- more specific Component typings for MDXComponent
- added assertion to json import
- dropped yarn.lock, and added packageManager field for corepack pnpm users

### CODE REVIEW

File: src/pages/Blog.data.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { RouteDataFunc } from 'solid-app-router';
 import type { Component } from 'solid-js';
 
-export type MDXComponent = Component<{ components: Record<string, Component> }>;
+export type MDXComponent = Component<{ components: Record<string, unknown> }>;
```
Comment (modderme123) [inline src/pages/Blog.data.ts:4]: Any reason we aren't just fully typing MDXComponent?

File: package.json [hunk 2]
```diff
@@ -81,5 +81,6 @@
     "vite": "^3.0.4",
     "vite-plugin-pwa": "^0.12.3",
     "vite-plugin-solid": "^2.3.0"
-  }
+  },
+  "packageManager": "pnpm@7.9.0"
```
Comment (modderme123) [inline package.json:85]: Given that we're adding packageManager, should we also add `engines: { "pnpm": ">=7.9.0" }`?

File: src/pages/Blog.data.ts (hunk 1 above)
Comment (high1) [reply src/pages/Blog.data.ts]: True, this is a much better solutions, although it requires for us to be specific about which components we do use.

File: package.json (hunk 2 above)
Comment (high1) [reply package.json:86]: Done in 0c4e7f04920fffb6d9dac74d24b884e2a45cabdb.

File: src/pages/BlogArticle.tsx [hunk 3]
```diff
@@ -45,15 +47,17 @@ export const BlogArticle: Component = () => {
                     <data.article
                       components={{
                         ListenNotesEpisode,
-                        Tweet: (props) => (
+                        Tweet: (props: TweetProperties) => (
```
Comment (modderme123) [inline src/pages/BlogArticle.tsx:50]: should be able to do `(props) => (` here (also does the linter warn about this?)

Comment (high1) [reply src/pages/BlogArticle.tsx]: Yes, you're right. Type inference makes this unneeded now. Removed in b18b9733a0b1dcea4753cbbd3bd60c247bbfb2d6.

**Comment by davedbase** [comment]
Thanks!

**Comment by modderme123** [comment]
Yes

**Pull request merged**